src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -557,20 +570,80 @@ def delete_ag_url_path_map_rule(resource_group_name, application_gateway_name, u
     ag = ncf.get(resource_group_name, application_gateway_name)
-    ag.web_application_firewall_configuration = \
-        ApplicationGatewayWebApplicationFirewallConfiguration(enabled == 'true', firewall_mode)
+    if supported_api_version(ResourceType.MGMT_NETWORK, min_api='2017-03-01'):  # pylint: disable=too-many-nested-blocks
+        ag.web_application_firewall_configuration = \
+            ApplicationGatewayWebApplicationFirewallConfiguration(
+                enabled == 'true', firewall_mode, rule_set_type, rule_set_version)
+        if disabled_rule_groups or disabled_rules:
+            ApplicationGatewayFirewallDisabledRuleGroup = get_sdk(
+                ResourceType.MGMT_NETWORK,
+                'ApplicationGatewayFirewallDisabledRuleGroup', mod='models')
+
+            disabled_groups = []
+
+            # disabled groups can be added directly
+            for group in disabled_rule_groups or []:
+                disabled_groups.append(ApplicationGatewayFirewallDisabledRuleGroup(group))
+
+            # for disabled rules, we have to look up the IDs
+            if disabled_rules:
+                results = list_ag_waf_rule_sets(
+                    ncf, _type=rule_set_type, version=rule_set_version, group='*')
+                for item in results:
+                    for group in item.rule_groups:
+                        disabled_group = ApplicationGatewayFirewallDisabledRuleGroup(
+                            group.rule_group_name, [])
+
+                        for rule in group.rules:
+                            if str(rule.rule_id) in disabled_rules:
+                                disabled_group.rules.append(rule.rule_id)
+                        if disabled_group.rules:
+                            disabled_groups.append(disabled_group)
+            ag.web_application_firewall_configuration.disabled_rule_groups = disabled_groups
+    else:
+        ag.web_application_firewall_configuration = \
+            ApplicationGatewayWebApplicationFirewallConfiguration(
+                enabled == 'true', firewall_mode)
+
     return ncf.create_or_update(resource_group_name, application_gateway_name, ag, raw=no_wait)
 
 def show_ag_waf_config(resource_group_name, application_gateway_name):
     return _network_client_factory().application_gateways.get(
         resource_group_name, application_gateway_name).web_application_firewall_configuration
 
+
+def list_ag_waf_rule_sets(client, _type=None, version=None, group=None):
+    results = client.list_available_waf_rule_sets().value
+    filtered_results = []
+    # filter by rule set name or version
+    for rule_set in results:
+        if _type and _type.lower() != rule_set.rule_set_type.lower():
+            continue
+        if version and version.lower() != rule_set.rule_set_version.lower():
+            continue
+
+        filtered_groups = []
+        for rule_group in rule_set.rule_groups:
+            if not group:
+                rule_group.rules = None
```

Comment:
This was a usability decision I made. If you don't specify a rule group to show rules for (or * to show for all groups) then I remove this because otherwise you'll be bombarded with pages of stuff. I've shown the example output for this command in each of the cases in a comment.